proposal/color-4-new-spaces-js.d.ts.md

```diff
@@ -218,68 +285,190 @@ colors as best they can, but it may be necessary in some cases.
+    h: keys.includes('h') ? options.h : internal.channel('h'),
+    alpha: keys.includes('alpha') ? options.alpha : internal.channel('alpha'),
+    space: space
+  })
+  ```
+
+* If `space` equals `a98-rgb`, `display-p3`, `prophoto-rgb`, `srgb`, or
+  `srgb-linear`, let `changedColor` be the result of:
+
+  ```js
+  SassColor({
+    red: keys.includes('red') ? options.red : internal.channel('red'),
+    green: keys.includes('green') ? options.green : internal.channel('green'),
+    blue: keys.includes('blue') ? options.blue : internal.channel('blue'),
+    alpha: keys.includes('alpha') ? options.alpha : internal.channel('alpha'),
+    space: space
+  })
+  ```
+
+* If `space` equals `xyz`,  `xyz-d50`, or `xyz-d65`, let `changedColor` be the
+    result of:
+
+  ```js
+  SassColor({
+    x: keys.includes('x') ? options.x : internal.channel('x'),
+    y: keys.includes('y') ? options.y : internal.channel('y'),
+    z: keys.includes('z') ? options.z : internal.channel('z'),
+    alpha: keys.includes('alpha') ? options.alpha : internal.channel('alpha'),
+    space: space
+  })
+  ```
+
+* Return the result of [`changedColor.toSpace(initialSpace)`].
+  
+[`internal.space()`]: #space
+[`internal.toSpace(space)`]: #tospace
+[`changedColor.toSpace(initialSpace)`]: #tospace
 
 ```ts
-changeChannels(
-  channels: {
-    [key in ChannelNameRGB]?: ChannelValue;
-  },
-  space?: ColorSpaceRGB
+change(
+  options: {
+    [key in ChannelName]?: ChannelValue;
+  } & {alpha?: number}
 ): SassColor;
 
-changeChannels(
-  channels: {
-    [key in ChannelNameHWB]?: ChannelValue;
-  },
-  space?: ColorSpaceHWB
+change(
+  options: {
+    [key in ChannelNameHSL]?: ChannelValue;
+  } & {
+    alpha?: number;
+    space: ColorSpaceHSL;
+  }
 ): SassColor;
 
-changeChannels(
-  channels: {
-    [key in ChannelNameHSL]?: ChannelValue;
-  },
-  space?: ColorSpaceHSL
+change(
+  options: {
+    [key in ChannelNameHWB]?: ChannelValue;
+  } & {
+    alpha?: number;
+    space: ColorSpaceHWB;
+  }
 ): SassColor;
 
-changeChannels(
-  channels: {
-    [key in ChannelNameXYZ]?: ChannelValue;
-  },
-  space?: ColorSpaceXYZ
+change(
+  options: {
+    [key in ChannelNameLAB]?: ChannelValue;
+  } & {
+    alpha?: ChannelValue;
+    space: ColorSpaceLAB;
+  }
 ): SassColor;
 
-changeChannels(
-  channels: {
+change(
+  options: {
     [key in ChannelNameLCH]?: ChannelValue;
-  },
-  space?: ColorSpaceLCH
+  } & {
+    alpha?: ChannelValue;
+    space: ColorSpaceLCH;
+  }
 ): SassColor;
 
-changeChannels(
-  channels: {
-    [key in ChannelNameLAB]?: ChannelValue;
-  },
-  space?: ColorSpaceLAB
+change(
+  options: {
+    [key in ChannelNameRGB]?: ChannelValue;
+  } & {
+    alpha?: ChannelValue;
+    space: ColorSpaceRGB;
+  }
+): SassColor;
+
+change(
+  options: {
+    [key in ChannelNameXYZ]?: ChannelValue;
+  } & {
+    alpha?: ChannelValue;
+    space: ColorSpaceXYZ;
+  }
 ): SassColor;
 ```
 
-### `interpolate()`
+#### `interpolate`
```

Comment:
This also needs to be rewritten in spec language

Reply:
Updated in [b42c323](https://github.com/sass/sass/pull/3665/commits/b42c3233343c48736f5a2a77e7dfaab7d1f64037)